openedx/openedx-platform
# feat: [FC-0056] Add calculation for percentage of the completions by …

### Description
This PR draws attention to the need to add the calculation of user progress to the API for navigating the sidebar that returns the course structure.

It is worth noting that the progress for sections is calculated by the state of the units' improvements in this section, and not directly by their subsections

### Testing instructions

- Run master devstack.
- Start platform make dev.up and make checkout on this branch.
- Create a course with different access rights to course blocks (sections/subsections/subsections) for different users.
- Make GET requests to the API (/api/course_home/v1/navigation/{course_id}) and check if the API returns the correct progress(`completion_stat` field) on the compliments for the user.
- Get new compliments in a test course
- Make GET requests to the API again and check that the progress has changed.
